File: native/ui_native/app/src/main/java/com/example/lab2_uinative/model/Concert.kt
package com.example.lab2_uinative.model

import android.os.Parcelable
import android.util.Log
import kotlinx.parcelize.Parcelize

@Parcelize
data class Concert(
    var name: String,
    var description: String,
    var date: org.threeten.bp.LocalDate,
    var location: String,
    var performers: List<String>,
    var id: Int
): Parcelable {

    companion object {
        var currentId = 0
    }

    constructor(name: String, description: String, date: org.threeten.bp.LocalDate,
                location: String, performers: List<String>) : this(name, description, date, location, performers, currentId++){
        Log.i("Concert Model: ", "CurrentId is $currentId")
    }

    fun formatPerformers(): String {
        return "Performers: ${performers.joinToString(", ")}"
    }

    fun formatPerformersEditForm(): String {
        return performers.joinToString(", ")
    }

    override fun toString(): String {
        return "$name - $description - ${date.toString()} - $location - ${formatPerformers()}"
    }
}

File: server/nonnative-server/src/model/concert.js
class Concert {
    constructor(id, name, description, date, location, performers) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.date = date;
      this.location = location;
      this.performers = performers;
    }
  
    static fromDbRow(row) {
        return new Concert(
          row.id,
          row.name,
          row.description,
          row.date,
          row.location,
          Array.isArray(row.performers) 
            ? row.performers 
            : row.performers.split(',').map((p) => p.trim())
        );
      }
  
    toDbRow() {
      return {
        id: this.id,
        name: this.name,
        description: this.description,
        date: this.date,
        location: this.location,
        performers: this.performers.join(', '),
      };
    }

    toJSON() {
        return {
          id: this.id,
          name: this.name,
          description: this.description,
          date: this.date,
          location: this.location,
          performers: Array.isArray(this.performers) ? this.performers : [],
        };
    }
}
  
  module.exports = Concert;